cloudsim-plus-testbeds/src/main/java/org/cloudsimplus/migration/ExampleMigrationVmWhenCpuMetricIsViolated.java

```diff
@@ -0,0 +1,340 @@
+import org.cloudsimplus.sla.readJsonFile.SlaMetric;
+import org.cloudsimplus.sla.readJsonFile.SlaReader;
+import org.cloudsimplus.util.tablebuilder.CloudletsTableBuilderHelper;
+
+/**
+ * This example shows how to do a migration using CpuUtilization threshold
+ * established in the SLA Contract.
+ * When the CPU metric is violated the migration is made.
+ * 
+ * @author raysaoliveira
+ */
+public class ExampleMigrationVmWhenCpuMetricIsViolated {
+
+    private static final int SCHEDULE_TIME_TO_PROCESS_DATACENTER_EVENTS = 5;
+    private static final double DATACENTER_COST_PER_CPU = 3.0;
+    private static final double DATACENTER_COST_PER_RAM = 0.05;
+    private static final double DATACENTER_COST_PER_STORAGE = 0.001;
+    private static final double DATACENTER_COST_PER_BW = 0.0;
+
+    private static final int HOST_MIPS_BY_PE = 1000;
+    private static final int HOST_NUMBER_OF_PES = 2;
+    private static final long HOST_RAM = 500000; //host memory (MB)
+    private static final long HOST_STORAGE = 1000000; //host storage
+    private static final long HOST_BW = 100000000L;
+
+    private static final int VM_MIPS = 1000;
+    private static final long VM_SIZE = 1000; //image size (MB)
+    private static final int VM_RAM = 10000; //vm memory (MB)
+    private static final long VM_BW = 100000;
+    private static final int VM_PES_NUM = 1; //number of cpus
+
+    private static final long CLOUDLET_LENGHT = 20000;
+    private static final long CLOUDLET_FILESIZE = 300;
+    private static final long CLOUDLET_OUTPUTSIZE = 300;
+    private static List<Cloudlet> list = new ArrayList<>();
+
+    /**
+     * The percentage of CPU that a cloudlet will use when it starts executing
+     * (in scale from 0 to 1, where 1 is 100%). For each cloudlet create, this
+     * value is used as a base to define CPU usage.
+     *
+     * @see #createAndSubmitCloudlets(DatacenterBroker)
+     */
+    private static final double CLOUDLET_INITIAL_CPU_UTILIZATION_PERCENTAGE = 0.6;
+
+    /**
+     * Defines the speed (in percentage) that CPU usage of a cloudlet will
+     * increase during the simulation tie. (in scale from 0 to 1, where 1 is
+     * 100%).
+     *
+     * @see #createAndSubmitCloudletsWithDynamicCpuUtilization(double, double,
+     * Vm, DatacenterBroker)
+     */
+    public static final double CLOUDLET_CPU_USAGE_INCREMENT_PER_SECOND = 0.05;
+
+    private static final int NUMBER_OF_HOSTS_TO_CREATE = 3;
+    private static final int NUMBER_OF_VMS_TO_CREATE = NUMBER_OF_HOSTS_TO_CREATE + 1;
+    private static final int NUMBER_OF_CLOUDLETS_TO_CREATE_BY_VM = 1;
+
+    private static final List<Vm> vmlist = new ArrayList<>();
+    private static CloudSim simulation;
+
+    /**
+     * The file containing the SLA Contract in JSON format.
+     */
+    private static final String METRICS_FILE = "/Users/raysaoliveira/Desktop/TeseMestradoEngInformatica/cloudsim-plus/cloudsim-plus-testbeds/src/main/java/org/cloudsimplus/sla/readJsonFile/SlaMetrics.json";
```

Comment:
It has to be provided a relative path to the json file.
Move the file to the "resources" directory inside the cloudsim-plus-testbeds/src/main/resources directory and define the value for such a constant as METRICS_FILE  = ResourceLoader.getResourcePath(VmMigrationWhenCpuMetricIsViolatedExample.class, "SlaMetrics.json");